internal/proxyservice/ssh/listener.go

```diff
@@ -0,0 +1,81 @@
+package ssh
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/go-ozzo/ozzo-validation"
+	"golang.org/x/crypto/ssh"
+
+	config_v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
+)
+
+// Listener accepts SSH connections and MITMs them using a ServiceConnector.
+//
+// NOTE: This MITM approach to SSH is experimental. The ssh-agent approach is
+// better validated and probably better all-around.
+
+// serviceHasCredentials validates that a service has all necessary credentials.
+type serviceHasCredentials struct {
```

Comment:
I think this name might need a better one - it sounds like it's a boolean but it's not

Reply:
wasn't needed anymore